plumbing/format/commitgraph/encoder.go

```diff
@@ -99,9 +100,9 @@ func (e *Encoder) prepare(idx Index, hashes []plumbing.Hash) (hashToIndex map[pl
 
 	// Find out if we will need extra edge table
 	for i := range len(hashes) {
-		v, _ := idx.GetCommitDataByIndex(uint32(i))
+		v, _ := idx.GetCommitDataByIndex(uint32(i)) //nolint:gosec // G115: i is loop index
```

Comment:
Slightly orthogonal, but let's take this opportunity and handle the `err` as that would mean `v` is then `nil`.
```suggestion
		v, err := idx.GetCommitDataByIndex(uint32(i)) //nolint:gosec // G115: i is loop index
```